tests/torch/sparsity/movement/test_components.py

```diff
@@ -0,0 +1,448 @@
+        ),
+        dict(
+            config={'mode': 'block', 'target_scopes': 'mock'},
+            error_info='Missing `sparse_factors`'
+        ),
+        dict(
+            config={'mode': 'block', 'sparse_factors': [2], 'target_scopes': 'mock'},
+            error_info='expects tuple of two'
+        ),
+        dict(
+            config={'mode': 'block', 'sparse_factors': [2, 2], 'axis': 0, 'target_scopes': 'mock'},
+            error_info='not expect specified `axis`'
+        ),
+        dict(
+            config={'mode': 'per_dim', 'target_scopes': 'mock'},
+            error_info='Missing `axis`'
+        ),
+        dict(
+            config={'mode': 'per_dim', 'axis': 0, 'sparse_factors': [1, 1], 'target_scopes': 'mock'},
+            error_info='not expect specified `sparse_factors`'
+        ),
+        dict(
+            config={'mode': 'per_dim', 'axis': 0},
+            error_info='Missing `target_scopes`'
+        )
+    ])
+    def test_error_on_creating_from_wrong_sparse_config_by_scope(self, desc: dict):
+        with pytest.raises(ValueError, match=desc['error_info']):
+            _ = SparseConfigByScope.from_config(desc['config'])
+
+
+desc_test_sparsifier_forward = {
+    'block': dict(
+        sparse_structure_by_scopes=[{'mode': 'block', 'sparse_factors': [2, 2], 'target_scopes': '{re}model'}],
+        init_weight_importance=torch.FloatTensor([[0, 1], [0, 1]]),
+        init_bias_importance=torch.FloatTensor([1, 0]),
+        ref_masked_weight=torch.FloatTensor([[0, 0, 2, 3], [0, 0, 6, 7], [0, 0, 10, 11], [0, 0, 14, 15]]),
+        ref_masked_bias=torch.FloatTensor([0, 1, 0, 0]),
+    ),
+    'per_row': dict(
+        sparse_structure_by_scopes=[{'mode': 'per_dim', 'axis': 0, 'target_scopes': '{re}model'}],
+        init_weight_importance=torch.FloatTensor([[0], [1], [0], [1]]),
+        init_bias_importance=torch.FloatTensor([1, 1, 0, 0]),
+        ref_masked_weight=torch.FloatTensor([[0] * 4, [4, 5, 6, 7], [0] * 4, [12, 13, 14, 15]]),
+        ref_masked_bias=torch.FloatTensor([0, 1, 0, 0]),
+    ),
+    'per_column': dict(
+        sparse_structure_by_scopes=[{'mode': 'per_dim', 'axis': 1, 'target_scopes': '{re}model'}],
+        init_weight_importance=torch.FloatTensor([0, 1, 0, 1]),
+        init_bias_importance=torch.FloatTensor([0]),
+        ref_masked_weight=torch.FloatTensor([[0, 1, 0, 3], [0, 5, 0, 7], [0, 9, 0, 11], [0, 13, 0, 15]]),
+        ref_masked_bias=torch.FloatTensor([0, 0, 0, 0]),
+    ),
+    'fine': dict(
+        sparse_structure_by_scopes=[{'mode': 'fine', 'sparse_factors': [1, 1], 'target_scopes': '{re}model'}],
+        init_weight_importance=torch.FloatTensor([[0, 1, 1, 1], [0, 1, 1, 1], [1] * 4, [0] * 4]),
+        init_bias_importance=torch.FloatTensor([0, 0, 0, 1]),
+        ref_masked_weight=torch.FloatTensor([[0, 1, 2, 3], [0, 5, 6, 7], [8, 9, 10, 11], [0] * 4]),
+        ref_masked_bias=torch.FloatTensor([0, 0, 0, 3]),
+    ),
+}
```

Comment:
Same - refactor to dataclass usage

Reply:
We decided to use dict, since it's needed just for a single test. Creation of many dataclasses seems to us unreasonable here.